### ISSUE TITLE
Question about addr handling

### ISSUE THREAD

**Issue by amitiuttarwar** [btcsuite/btcd#1709]
Hi! I'm proposing some changes to `addr` handling in Bitcoin Core, and trying to ensure there will not be negative consequences for other clients on the network. For more information about the changes, please see my [mailing list post](https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2021-April/018784.html). The crux is that, if the changes are accepted, bitcoin core would stop sending `addr` messages to peers who don't "indicate interest" aka initiate an addr-related message (addr, addrv2, getaddr, or sendaddrv2). 

So, I'd like to ensure that btcd clients will initiate an addr related message to outbound peers. I've looked in the code and believe the software plays nice, and am looking for a +1 from someone more familiar with this project. 

In `server.go`, I believe the `peerHandler()` calls `handleAddPeerMsg` when the node connects to a new peer ([link](https://github.com/btcsuite/btcd/blob/2d7825cf709fc6ac15921cecfd4d62ec78ccbba2/server.go#L2169)). The logic of that function seems like if it successfully connects to an outbound peer on mainnet, it will send an `addr` message to self advertise, and sometimes send a `getaddr` message ([link](https://github.com/btcsuite/btcd/blob/2d7825cf709fc6ac15921cecfd4d62ec78ccbba2/server.go#L1694)). Is this an accurate interpretation? 

If my understanding is right, there should be no compatibility issues with my proposed changes.

**tldr; I'm proposing a change to Bitcoin Core and want to make sure it does not break btcd, can someone confirm my understanding of the code?**

Thanks in advance! 
Amiti

**Comment by Roasbeef** [comment]
> So, I'd like to ensure that btcd clients will initiate an addr related message to outbound peers

Yeah your reading of the code is correct, we'll always send a `getaddr` to newly connected outbound peers, unless its simnet, or we aren't listening ourselves on a public network interface.

**Comment by amitiuttarwar** [comment]
awesome! thank you :)
